src/functions/JpowerIntegrals.h

```diff
@@ -0,0 +1,151 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with MRCPP.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * For information on the complete list of contributors to MRCPP, see:
+ * <https://mrcpp.readthedocs.io/>
+ */
+
+#pragma once
+
+#include <iostream>
+#include <complex>
+#include <vector>
+#include <cmath>
+
+namespace mrcpp {
+
+/** @class JpowerIntegrals
+ *
+ * @brief A class needed for construction Schrodinger time evolution operator
+ *
+ * @details A two dimensional array consisting of integrals $J_m$ as follows.
+ * Our main operator has the following expansion
+\begin{equation}
+\label{operator_correlation_expansion}
+    \left[ \sigma_l^{\mathfrak n} \right]_{pj}
+    (a)
+    =
+    \sum_{k = 0}^{\infty}
+    C_{jp}^{2k}
+    \widetilde J_{2k + j + p}(l, a)
+    ,
+\end{equation}
+where $a = t \mathfrak N^2 = t 4^{\mathfrak n}$
+and
+\begin{equation}
+\label{power_integral}
+    \widetilde J_m
+    =
+    \frac
+    {
+        I_m
+        e^{ i \frac {\pi}4 (m - 1) }
+    }
+    {
+        2 \pi ( m + 2 )!
+    }
+    =
+    \frac
+    {
+        e^{ i \frac {\pi}4 (m - 1) }
+    }
+    {
+        2 \pi ( m + 2 )!
+    }
+    \int_{\mathbb R}
+    \exp
+    \left(
+        \rho l \exp \left( i \frac \pi 4 \right) - a \rho^2
+    \right)
+    \rho^m
+    d \rho
+\end{equation}
+satisfying the following relation
+\begin{equation}
+\label{power_integral_recursion}
+    \widetilde J_{m+1}
+    =
+    \frac
+    {
+        il
+    }
+    {
+        2a (m + 3)
+    }
+    \widetilde J_m
+    +
+    \frac {im}{2a(m + 2)(m + 3)}
+    \widetilde J_{m-1}
+    =
+    \frac
+    {
+        i
+    }
+    {
+        2a (m + 3)
+    }
+    \left(
+        l
+        \widetilde J_m
+        +
+        \frac {m}{(m + 2)}
+        \widetilde J_{m-1}
+    \right)
+    , \quad
+    m = 0, 1, 2, \ldots,
+\end{equation}
+with $\widetilde J_{-1} = 0$ and
+\begin{equation}
+\label{power_integral_0}
+    \widetilde J_0
+    =
+    \frac{ e^{ -i \frac{\pi}4 } }{ 4 \sqrt{ \pi a } }
+    \exp
+    \left(
+        \frac{il^2}{4a}
+    \right)
+    .
+\end{equation}
+ *
+ *  
+ */
+class JpowerIntegrals
+{
+public:
+    /// @brief creates an array of power integrals
+    /// @param a : parameter a
+    /// @param scaling : scaling level
+    /// @param M : maximum amount of integrals for each l
+    /// @param treshold : lower limit for neglecting the integrals
+    /// @details The array is orginised as a vector ordered as \f$l = 0, 1, 2, \ldots, 2^n - 1, 1 - 2^n, 2 - 2^n, \ldots, -2, -1 \f$.
+    JpowerIntegrals(double a, int scaling, int M, double treshold = 1.0e-15);
+    JpowerIntegrals(const JpowerIntegrals& other);
+
+    ~JpowerIntegrals();
+
+    int scaling;  //it is probably not used
+    std::vector<std::vector<std::complex<double>>> integrals;
```

Comment:
I suggest you use a `Eigen::MatrixXcd` (https://eigen.tuxfamily.org/dox/group__TutorialMatrixClass.html) instead of a doubly-nested `std::vector` here:
* Eigen allocates the matrix data in one chunk, and I believe you know the number of rows and columns when you start computing the integrals you need.
* The API of the Eigen matrices is more intuitive for index-based operations.
* Eigen can be faster than using `std::vector` directly.